lib/Sema/CSDiagnostics.cpp

```diff
@@ -32,7 +32,9 @@
 #include "swift/AST/Stmt.h"
 #include "swift/AST/Types.h"
 #include "swift/Basic/SourceLoc.h"
+#include "swift/Basic/StringExtras.h"
 #include "swift/Parse/Lexer.h"
+#include "swift/Sema/IDETypeChecking.h"
```

Comment:
OK, I will fix it.

It is included to use `getOriginalArgumentList` helper function at `SingleLabelingFailure::diagnoseAsError`.
It integrates `ParenExpr`, `TupleExpr`, and other cases.

I don't know well what AST may comes here.
I will be looking into it.

Reply:
I'd suggest to avoid dealing with AST at all and record all necessary information in the fix/diagnostic. We are hoping that we'd be able to model argument in calls differently in the future so adjusting diagnostics proactively means less refactoring in the future.

Reply:
Oh I never thought about not touching AST.
OK!
Thank for telling me future direction.